### PR TITLE
Provide admins with a way to regenerate their API token

### LINKED ISSUE #2410: Allow Users to Refresh their API Token
Add a button on the User Profile page that generates a new API token for the user.

### PR DESCRIPTION
**Pull request by briri**
Fixes #2410 .

Added ability for user to regenerate their own API token (or create one if it was not initialized for some reason when they became an Org admin). Also updated the application.js to make jQuery and the notificationHelper functions available within the js.erb files.

Changes proposed in this PR:
- Move API token and link to wiki documentation to its own tab on Edit Profile page
- Update initializer to provide both v0 and v1 API urls
- Add a 'Regenerate token' button to new tab and brief explanation of v0 and v1 with links to their wiki pages
- Add route, controller and js.erb to regerate the user's token
- Update User model to separate out the create token functionality from the create_or_update method
- Update the application.js to make jQuery and the renderNotice and renderAlert helper functions available to the js.erb files
- Update api_client credentials mailer to use updated initializer values
- Update the refresh api client credentials js.erb

### CODE REVIEW

File: app/views/super_admin/api_clients/email_credentials.js.erb [hunk 1]
```diff
@@ -1,6 +1,2 @@
 var msg = '<%= _("The credentials have been sent to %{email}.") % { email: @api_client.contact_email } %>';
-
-<%# TODO: replace this with the notificationHelper.js once we move to Rails 5 %>
-var notification = document.getElementById("notification-area");
-notification.append(msg);
-notification.classList.remove('hide');
\ No newline at end of file
+renderNotice(msg);
```
Comment (xsrust) [inline app/views/super_admin/api_clients/email_credentials.js.erb]: Great to see these TODO's going away:)

**Review by xsrust — CHANGES_REQUESTED** [review]

File: app/policies/user_policy.rb [hunk 2]
```diff
@@ -59,6 +59,10 @@ def acknowledge_notification?
     true
   end
 
+  def refresh_token?
+    true
+  end
```
[the remark this answers is not in the document]
Comment (briri) [reply app/policies/user_policy.rb:64]: Yes, the button should not appear unless they're an Org admin but it would make sense to add logic to the policy

**Comment by briri** [comment]
Ok @xsrust I added some permission checks to that policy

**Review by xsrust — APPROVED** [review]
Thanks for updating

**Pull request merged**